plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/credential/CredentialPropertiesProvider.java

```diff
@@ -22,5 +22,5 @@
  */
 public interface CredentialPropertiesProvider
```

Comment:
Is it possible to make this generic like `CredentialPropertiesProvider<K, V>` to preserve the type-safety?